app/assets/javascripts/components/timeline/block.jsx

```diff
@@ -37,7 +37,7 @@ const Block = createReactClass({
   },
 
   passedUpdateBlock(selectedIds) {
-    const newBlock = $.extend(true, {}, this.props.block);
+    const newBlock = structuredClone(this.props.block);
```

Comment:
Do we have polyfills to ensure that this works across browsers? Browser support for this just dates to early 2022, and elsewhere in the codebase we use `Object.assign` for this kind of thing.

Reply:
Yeah `Object.assign` would be simpler but it doesn't clone the object recursively(which the JQuery version does)

`core-js` does have polyfills for [`structuredClone`](https://github.com/zloirock/core-js#structuredclone) so it shouldn't be an issue.

Reply:
Hmm... do we even need deep cloning here? I can't think of a reason we would.

Reply:
No idea :shrug: I went with it just to be safe. Should I just use `Object.assign`?

Reply:
I think so. I'm pretty confident there's no reason why a shallow clone wouldn't be fine here. The `block` object is not that complicated, and here we're ultimately just replacing a block (as identified by its `id` attribute) in the redux state with a version of it that has been modified in an explicit way.